drivers/hopwa_caper/app/models/hopwa_caper/enrollment.rb

```diff
@@ -0,0 +1,140 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HopwaCaper
+  class Enrollment < GrdaWarehouseBase
+    include SqlHelper
+    self.table_name = 'hopwa_caper_enrollments'
+
+    has_many :hud_reports_universe_members,
+             -> do
+               where(::HudReports::UniverseMember.arel_table[:universe_membership_type].eq('HopwaCaper::Enrollment'))
+             end,
+             inverse_of: :universe_membership,
+             class_name: 'HudReports::UniverseMember',
+             foreign_key: :universe_membership_id
+    has_many :services, class_name: 'HopwaCaper::Service', primary_key: :enrollment_id
+
+    def self.as_report_members
+      all.map do |record|
```

Comment:
Isn't the `all` here going to override any calling scope?  
Should this start with something like `current_scope`?  Can you do that in a class method?

Reply:
hmm, I don't think [scope.all()](https://github.com/rails/rails/blob/fc734f28e65ef8829a1a939ee6702c1f349a1d5a/activerecord/lib/active_record/scoping/named.rb#L22) overrides the calling scope. It's very similar to calling current_scope. I can use current_scope instead if you think it would be more clear

Reply:
Interesting.  if `current_scope` works, that's what we've done in the past, so I guess that's preferable, (and wouldn't cause me to think twice about it.)  If it's not available, this is fine as is.